File: UI/src/app/models/Group.ts
import { UserWithBalance } from "./UserWithBalance";

export interface Group {
  groupId: number;  // Make groupId optional
  name: string;
  description: string;
  createdDate: Date;
  members?: UserWithBalance[]; 
}

export interface GroupCreation {
  name: string;
  description: string;
  createdDate: Date;
}

File: Expense-Sharing-Application-API/Expense-Sharing-Application-API/Data_Access_Layer/Models/Group.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Data_Access_Layer.Models
{
    public class Group
    {
        public int GroupId { get; set; }

        [Required(ErrorMessage = "Name is required.")]
        [StringLength(50, ErrorMessage = "Name cannot exceed 50 characters.")]
        public string? Name { get; set; }

        [StringLength(200, ErrorMessage = "Description cannot exceed 200 characters.")]
        public string? Description { get; set; }

        [Required(ErrorMessage = "Date is required.")]
        [DataType(DataType.DateTime)]
        public DateTime CreatedDate { get; set; }

        [JsonIgnore]
        public ICollection<GroupMember>? Members { get; set; }

        [JsonIgnore]
        public ICollection<Expense>? Expenses { get; set; }
    }
}
